pkg/scheduler/ugm/queue_tracker.go

```diff
@@ -256,16 +259,34 @@ func (qt *QueueTracker) headroom(hierarchy []string) *resources.Resource {
 		if qt.childQueueTrackers[childName] == nil {
 			qt.childQueueTrackers[childName] = newQueueTracker(qt.queuePath, childName)
 		}
-		childHeadroom = qt.childQueueTrackers[childName].headroom(hierarchy[1:])
+		childHeadroom = qt.childQueueTrackers[childName].headroom(hierarchy[1:], trackType)
 	}
 	// arrived at the leaf or on the way out: check against current max if set
 	if !resources.Equals(resources.NewResource(), qt.maxResources) {
 		headroom = qt.maxResources.Clone()
 		headroom.SubOnlyExisting(qt.resourceUsage)
 		log.Log(log.SchedUGM).Debug("Calculated headroom",
 			zap.String("queue path", qt.queuePath),
+			zap.Int("tracking type", int(trackType)),
 			zap.Stringer("max resource", qt.maxResources),
 			zap.Stringer("headroom", headroom))
+	} else if resources.Equals(nil, childHeadroom) {
+		// Only user wildcard case gets to this block, so we only check user tracking type here.
+		// All non-specific groups get into a same wildcard cumulative group limit.
+		// The maxResources in wildcard group limit is not empty. Its qt.maxResources is not empty,
+		// so it gets calculated headroom in the upper block.
+		// ref: https://docs.google.com/document/d/1yCZSoHTrsistHktmPwgreY05u26eI35XwOYdHWP0LrQ/edit#heading=h.2zonl4qy889f
+		if trackType == user {
```

Comment:
Please use the below comment:

fall back on wild card user limit settings to calculate headroom only for unnamed users. For unnamed groups, "*" group tracker object would be used using the above if block to calculate headroom because resource usage added together for all unnamed groups under "*" group tracker object.

Reply:
Updated comment. I change few words for better understanding. Thanks for the review.

Reply:
I don't think comment has been updated. Providing links to the design doc and all is unnecessary. At times, we keep changing the implementation based on the situations.

Just replace it with the below comment:

// Fall back on wild card user limit settings to calculate headroom only for unnamed users. 
// For unnamed groups, "\*" group tracker object would be used using the above if block to calculate headroom 
// because resource usage added together for all unnamed groups under "\*" group tracker object.

Reply:
Okay. I will update it. Just curious, is "if" a redundant work in the sentence?

// For unnamed groups, "*" group tracker object would be used using the above "if" block to calculate headroom